includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,481 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+} // Exit if accessed directly
+
+class WC_Swedbank_Plugin {
+
+	/** Payment IDs */
+	const PAYMENT_METHODS = [
+		'payex_checkout',
+		'payex_psp_cc',
+		'payex_psp_invoice',
+		'payex_psp_vipps',
+		'payex_psp_swish'
+	];
+
+	/**
+	 * @var WC_Background_Swedbank_Pay_Queue
+	 */
+	public static $background_process;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		// Includes
+		$this->includes();
+
+		// Activation
+		register_activation_hook( __FILE__, [ $this, 'install' ] );
+
+		// Actions
+		add_filter( 'plugin_action_links_' . plugin_basename( __FILE__ ), [ $this, 'plugin_action_links' ] );
+		add_action( 'plugins_loaded', [ $this, 'init' ], 0 );
+		add_action( 'woocommerce_init', [ $this, 'woocommerce_init' ] );
+		add_action( 'woocommerce_loaded', [ $this, 'woocommerce_hook_loaded' ] );
+
+		// Add statuses for payment complete
+		add_filter( 'woocommerce_valid_order_statuses_for_payment_complete', [ $this, 'add_valid_order_statuses' ], 10, 2 );
+
+		// Status Change Actions
+		add_action( 'woocommerce_order_status_changed', __CLASS__ . '::order_status_changed', 10, 4 );
+
+		// Add meta boxes
+		add_action( 'add_meta_boxes', __CLASS__ . '::add_meta_boxes' );
+
+		// Add action buttons
+		add_action( 'woocommerce_order_item_add_action_buttons', __CLASS__ . '::add_action_buttons', 10, 1 );
+
+		// Add scripts and styles for admin
+		add_action( 'admin_enqueue_scripts', __CLASS__ . '::admin_enqueue_scripts' );
+
+		// Add Admin Backend Actions
+		add_action( 'wp_ajax_swedbank_pay_capture', [ $this, 'ajax_swedbank_pay_capture' ] );
+
+		add_action( 'wp_ajax_swedbank_pay_cancel', [ $this, 'ajax_swedbank_pay_cancel' ] );
+
+		// Filters
+		add_filter( 'swedbank_pay_generate_uuid', [ $this, 'generate_uuid' ], 10, 1 );
+		add_filter( 'swedbank_pay_payment_description', [ $this, 'payment_description' ], 10, 2 );
+
+		// Process swedbank queue
+		if ( ! is_multisite() ) {
+			add_action( 'customize_save_after', [ $this, 'maybe_process_queue' ] );
+			add_action( 'after_switch_theme', [ $this, 'maybe_process_queue' ] );
+		}
+
+		// Add admin menu
+		add_action( 'admin_menu', [ $this, 'admin_menu' ], 99 );
+
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ["require_once" statement detected. File manipulations are discouraged. Statement is not a function, no parentheses are required. Concatenating is forbidden.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)